sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -75,21 +84,33 @@ interface DefaultCredentialConstructor {
  * @internal
  */
 export class DefaultManagedIdentityCredential extends ManagedIdentityCredential {
+  // Constructor overload with just client id options
+  constructor(options?: DefaultAzureCredentialClientIdOptions);
+  // Constructor overload with just resource id options
+  constructor(options?: DefaultAzureCredentialResourceIdOptions);
+  // Constructor overload with just the other default options
+  // Last constructor overload with Union of all options not required since the above two constructor overloads have optional properties
   constructor(options?: DefaultAzureCredentialOptions) {
     const managedIdentityClientId =
       (options as DefaultAzureCredentialClientIdOptions)?.managedIdentityClientId ??
       process.env.AZURE_CLIENT_ID;
-    const managedResourceId = (options as DefaultAzureCredentialOptionsWithResourceId)
+    const managedResourceId = (options as DefaultAzureCredentialResourceIdOptions)
       ?.managedIdentityResourceId;
 
     // ManagedIdentityCredential throws if both the resourceId and the clientId are provided.
-    const managedIdentityOptions: ManagedIdentityCredentialOptions = {
-      resourceId: managedResourceId,
-      clientId: managedIdentityClientId,
-      ...options,
-    };
-
-    super(managedIdentityOptions);
+    if (managedResourceId) {
+      const managedIdentityResourceIdOptions: ManagedIdentityCredentialResourceIdOptions = {
+        ...options,
+        resourceId: managedResourceId,
+      };
+      super(managedIdentityResourceIdOptions);
+    } else if (managedIdentityClientId) {
+      const managedIdentityClientOptions: ManagedIdentityCredentialClientIdOptions = {
+        ...options,
+        clientId: managedIdentityClientId,
+      };
+      super(managedIdentityClientOptions);
+    }
```

Comment:
Need an `else super(something);` here to make sure we call super in cases where neither client ID nor resource ID is available.